### PR TITLE
UseNameofInPlaceOfString: Remove redundant null-check

### LINKED ISSUE #3023: UseNameofInPlaceOfString: Add coverage for early return
📝 We should make sure this early return line of the analyzer is still covered by tests:

https://github.com/dotnet/roslyn-analyzers/blob/1fa5f5c1167500b93bbebc301f240edde60effce/src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs#L64-L67

_Originally posted by @sharwell in https://github.com/dotnet/roslyn-analyzers/pull/3005_

### PR DESCRIPTION
**Pull request by Evangelink**
Fix #3023

### CODE REVIEW

File: src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs [hunk 1]
```diff
@@ -61,11 +61,6 @@ private void AnalyzeArgument(OperationAnalysisContext context)
                 return;
             }
 
-            if (argument.Parameter == null)
```
Comment (Evangelink) [inline src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs]: The hover popup says that `Parameter` can't be null so I am removing this un-needed/unreachable early exit.

Comment (sharwell) [reply src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs]: The hover popups are not a guarantee. In this case, the definition of `IArgumentOperation` is null oblivious.

Comment (Evangelink) [reply src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs]: Ah too bad! I thought that was something based on nullability declaration,

**Comment by Evangelink** [comment]
@sharwell do you know a way to produce a null parameter symbol from a unit-test?

**Comment by sharwell** [comment]
I do not.

**Comment by sharwell** [comment]
My first guess would be a call site that passes more arguments than the method has parameters.

**Comment by Evangelink** [comment]
From the various tests I have made, it seems that the `OperationKind.Argument` callback only occurs when a valid overload is found so I am really not sure what could be done to have something null. I'll close this PR except if you prefer that I add a `Debug.Assert` and handle the nullability in the switch on the following lines.

File: src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.CodeQuality.Analyzers/Core/Maintainability/UseNameofInPlaceOfString.cs]: I believe `IArgumentOperation.Parameter` cannot be null. @333fred to confirm.

**Comment by sharwell** [comment]
I would prefer to leave it as-is for now, but consider revisiting once we make more progress on the nullability annotations in the IOperation APIs (actively being worked on).

**Pull request closed without merging**